internal/storage/driver/memory/operation.go

```diff
@@ -190,7 +190,47 @@ func (s *operations) ListOperationsByInstanceID(instanceID string) ([]internal.O
 	return operations, nil
 }
 
-func (s *operations) ListOperationsInTimeRange(time.Time, time.Time) ([]internal.Operation, error) {
+func (s *operations) ListOperationsByInstanceIDGroupByType(instanceID string) (*internal.GroupedOperations, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	grouped := internal.GroupedOperations{
+		ProvisionOperations:      make([]internal.ProvisioningOperation, 0),
+		DeprovisionOperations:    make([]internal.DeprovisioningOperation, 0),
+		UpgradeKymaOperations:    make([]internal.UpgradeKymaOperation, 0),
+		UpgradeClusterOperations: make([]internal.UpgradeClusterOperation, 0),
+		UpdateOperations:         make([]internal.UpdatingOperation, 0),
+	}
+
+	for _, op := range s.operations {
+		switch op.Type {
+		case internal.OperationTypeProvision:
+			grouped.ProvisionOperations = append(grouped.ProvisionOperations, internal.ProvisioningOperation{Operation: op})
+
+		case internal.OperationTypeDeprovision:
+			grouped.DeprovisionOperations = append(grouped.DeprovisionOperations, internal.DeprovisioningOperation{Operation: op})
+
+		case internal.OperationTypeUpgradeKyma:
+			grouped.UpgradeKymaOperations = append(grouped.UpgradeKymaOperations, internal.UpgradeKymaOperation{Operation: op})
+
+		case internal.OperationTypeUpgradeCluster:
+			grouped.UpgradeClusterOperations = append(grouped.UpgradeClusterOperations, internal.UpgradeClusterOperation{Operation: op})
+
+		case internal.OperationTypeUpdate:
+			grouped.UpdateOperations = append(grouped.UpdateOperations, internal.UpdatingOperation{Operation: op})
+		}
```

Comment:
Could we log if there is type not matching any of these?

Reply:
Corrected in https://github.com/kyma-project/kyma-environment-broker/pull/514/commits/3b4a8b8200c665f42373a2dd3172cd17c09df5b5. I have added an error for postgresql DAO and panic for in memory implementation.